conda_build/render.py

```diff
@@ -217,9 +218,18 @@ def find_pkg_dir_or_file_in_pkgs_dirs(pkg_dist, m, files_only=False):
             with tarfile.open(pkg_file, 'w:bz2') as archive:
                 for entry in os.listdir(pkg_dir):
                     archive.add(os.path.join(pkg_dir, entry), arcname=entry)
-            pkg_subdir = os.path.join(m.config.croot, m.config.host_subdir)
+
+            # use the package's subdir
+            try:
+                info = json.loads(Path(pkg_dir, "info", "index.json").read_text())
+                subdir = info["subdir"]
+            except (FileNotFoundError, KeyError):
+                subdir = m.config.host_subdir
```

Comment:
I dislike this fallback, what do we do instead? let the errors propagate up?

Reply:
old packages don't have subdir IIRC